### PR TITLE
Changed the website to next.js application

### PR DESCRIPTION
**Pull request by naveen13-06**
Please review and suggest changes.

### CODE REVIEW

**Review by SaptarshiSarkar12 — DISMISSED** [review]
@naveen13-06 Please check if you are using Typescript as I can find a redundant `tsconfig.json` file. Also delete **next**, **thirteen** and **vercel** _svg_ files. Please also remove redundant **download section** from the main page. More changes in the performance needs to be made. Please make those changes.

**Comment by SaptarshiSarkar12** [comment]
Also, the social links are not redirecting to any external URL. Please add such features to them, the social links can be viewed in the footer of the present website.

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
Please solve the below errors
```
Warning: Invalid DOM property `class`. Did you mean `className`?
    at i
    at a
    at div
    at div
    at div
    at Contribute (webpack-internal:///./components/Contribute.js:17:23)
    at Home (webpack-internal:///./pages/index.js:112:30)
    at App (webpack-internal:///./pages/_app.js:11:16)
    at StyleRegistry (/home/saptarshi/Test new website/Website/node_modules/styled-jsx/dist/index/index.js:449:36)
    at PathnameContextProviderAdapter (/home/saptarshi/Test new website/Website/node_modules/next/dist/shared/lib/router/adapters.js:60:11)
    at AppContainer (/home/saptarshi/Test new website/Website/node_modules/next/dist/server/render.js:290:29)
    at AppContainerWithIsomorphicFiberStructure (/home/saptarshi/Test new website/Website/node_modules/next/dist/server/render.js:326:57)
    at div
    at Body (/home/saptarshi/Test new website/Website/node_modules/next/dist/server/render.js:613:21)
Warning: Invalid DOM property `fill-rule`. Did you mean `fillRule`?
    at path
    at svg
    at a
    at div
    at div
    at footer
    at div
    at Footer
    at Home (webpack-internal:///./pages/index.js:112:30)
    at App (webpack-internal:///./pages/_app.js:11:16)
    at StyleRegistry (/home/saptarshi/Test new website/Website/node_modules/styled-jsx/dist/index/index.js:449:36)
    at PathnameContextProviderAdapter (/home/saptarshi/Test new website/Website/node_modules/next/dist/shared/lib/router/adapters.js:60:11)
    at AppContainer (/home/saptarshi/Test new website/Website/node_modules/next/dist/server/render.js:290:29)
    at AppContainerWithIsomorphicFiberStructure (/home/saptarshi/Test new website/Website/node_modules/next/dist/server/render.js:326:57)
    at div
    at Body (/home/saptarshi/Test new website/Website/node_modules/next/dist/server/render.js:613:21)
Warning: Invalid DOM property `clip-rule`. Did you mean `clipRule`?
    at path
    at svg
    at a
    at div
    at div
    at footer
    at div
    at Footer
    at Home (webpack-internal:///./pages/index.js:112:30)
    at App (webpack-internal:///./pages/_app.js:11:16)
    at StyleRegistry (/home/saptarshi/Test new website/Website/node_modules/styled-jsx/dist/index/index.js:449:36)
    at PathnameContextProviderAdapter (/home/saptarshi/Test new website/Website/node_modules/next/dist/shared/lib/router/adapters.js:60:11)
    at AppContainer (/home/saptarshi/Test new website/Website/node_modules/next/dist/server/render.js:290:29)
    at AppContainerWithIsomorphicFiberStructure (/home/saptarshi/Test new website/Website/node_modules/next/dist/server/render.js:326:57)
    at div
    at Body (/home/saptarshi/Test new website/Website/node_modules/next/dist/server/render.js:613:21)
Warning: Invalid DOM property `crossorigin`. Did you mean `crossOrigin`?
    at script
    at head
    at Head (webpack-internal:///./node_modules/next/dist/pages/_document.js:279:1)
    at html
    at Html (webpack-internal:///./node_modules/next/dist/pages/_document.js:678:104)
    at Document
```

**Comment by SaptarshiSarkar12** [comment]
@naveen13-06 Also, remove the Download section from the main starting page

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
Please resolve the below error
```
Warning: Invalid DOM property `crossorigin`. Did you mean `crossOrigin`?
    at script
    at head
    at Head (webpack-internal:///./node_modules/next/dist/pages/_document.js:279:1)
    at html
    at Html (webpack-internal:///./node_modules/next/dist/pages/_document.js:678:104)
    at Document
```

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@naveen13-06 Please fix the `stats` section and remove redundant apostrophes from `Lets connect` part.

image

**Review by SaptarshiSarkar12 — DISMISSED** [review]
@naveen13-06 Please fix the colours in the footer --- when I click on the socials icons, they become white and invisible. Please make them visible properly.

image

---
Also, make the whole **Contact Us** section centered in the middle. And also change the colour and styling of **Send Message** button.

image

**Comment by naveen13-06** [comment]
> @naveen13-06 Please fix the `stats` section and remove redundant apostrophes from `Lets connect` part.
> 
> image

Hey @SaptarshiSarkar12 , i couldn't find the reason why the stats image is getting displayed like this. I have just imported the image from the url and there is no error while importing. As a work around i have downloaded it and added it. I know its not a permanent solution, but i couldnt find the reason for that

**Comment by SaptarshiSarkar12** [comment]
> > @naveen13-06 Please fix the `stats` section and remove redundant apostrophes from `Lets connect` part.
> > image
> 
> Hey @SaptarshiSarkar12 , i couldn't find the reason why the stats image is getting displayed like this. I have just imported the image from the url and there is no error while importing. As a work around i have downloaded it and added it. I know its not a permanent solution, but i couldnt find the reason for that

Okay. I have understood.
# Things to fix
- Animation overflows the screen in mobiles.
- Download buttons appears in the menu for mobiles
- Add pading to the container of images of contributors.

> **NOTE** - I am using Samsung Galaxy J2.

@naveen13-06 Please fix these errors

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
# Things to fix :
- The top portion of the menu when opened in menu, does not cover the top visible portion of the screen. Hence, the colour of the background is visible. So, it is recommended to **move the top margin of the menu to the topmost visible portion of the mobile screen.**
- There is a huge gap between the pictures of the stats in **About** page. Please fix that.  
   image
- The tagline in the About page is not centered and not in bold letters. If last one or two words cannot be fitted in a single sentence in the screens (especially of mobiles), then those words need to centered. Please fix that.  
  image
- The buttons below the collection of images of the contributors, should be centered with respect to the container of the images.  
  image
- Huge gaps are present in between the download buttons (I found this in case of desktops/laptops). Please fix that.  
  image
- In the **contact us section**, when only **name**, **mobile no.** and **email address** has been given, the **Send Message** button takes the user to the website for submitting email although **NO MESSAGE** has been written. This can lead to spams. So, please fix this as well.
- The colours of each page are not consistent. For example :
   - The header section of the download page contains blue colour but the colour of the rest of the page is white.
   - Similar case is with the contribute page where the rest of the page is dark blue but the header is very light blue.

  Please fix them. If you are ambiguous for choosing colours, you can watch the **60-30-10 rule**. Here is a reference video for the same - https://youtu.be/UWwNIMHFdW4
- Change the Image component of next for the **stats image** where a _static picture_ has been added (**[in this comment](https://github.com/SaptarshiSarkar12/Drifty/pull/204#issuecomment-1486126815)**). I think adding a dynamic sizing of teh image might help. Here is a reference video for the same - https://youtu.be/2U7yZ3wvFBM?t=840

### Please fix the changes and let me know @naveen13-06 .

**Comment by SaptarshiSarkar12** [comment]
Hey @naveen13-06 , I am busy now. So, I shall review your changes by tomorrow.

**Comment by naveen13-06** [comment]
> Hey @naveen13-06 , I am busy now. So, I shall review your changes by tomorrow.

It's ok, you can take your time

File: Website/pages/about.js [hunk 1]
```diff
@@ -22,8 +22,8 @@ export default function about(){
         </div>
         <div className="mt-5 mx-5 border-t-2 border-black">
         <h1 className="text-center font-bold text-3xl py-5">My Stats</h1>
-        <div className="grid grid-cols-2 space-x-2 justify-items-center py-5 pb-20">
-        <Image width={360} height={25}  src="/Resources/github-stats-1.svg" alt=""/>
+        <div className="grid grid-cols-2  lg:-space-x-44 justify-items-center py-5 pb-20">
+        <Image width={360} height={25} className="lg:ml-20"  src="/Resources/github-stats-1.svg" alt=""/>
```
Comment (SaptarshiSarkar12) [inline Website/pages/about.js:26]: Please try to change the stats image with a dynamic image link. You can refer here - https://youtu.be/2U7yZ3wvFBM?t=840 .

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
# Things to fix
- The colors of the pages like **Contribute**, etc. are not consistent with the whole theme of the website. 
   image
- The color of header of the **Download** page is not consistent with the whoole theme of the website.   
   image
- Add dark mode feature to the website.

File: Website/pages/about.js (hunk 1 above)
Comment (naveen13-06) [reply Website/pages/about.js:26]: I have tried adding the dynamic image link but still the image is not showing properly. Its the same as [this comment](https://github.com/SaptarshiSarkar12/Drifty/pull/204#issuecomment-1486126815)

**Comment by naveen13-06** [comment]
> # Things to fix
> * The colors of the pages like **Contribute**, etc. are not consistent with the whole theme of the website.
>   image
> * The color of header of the **Download** page is not consistent with the whoole theme of the website.
>   image
> * Add dark mode feature to the website.

Sorry, but i dont know much about how to choose colors or design it. If u want u could create a new issue for it or if you have time you can provide the design and color for each elements and i will replace the color there.

**Comment by SaptarshiSarkar12** [comment]
> > # Things to fix
> > 
> > * The colors of the pages like **Contribute**, etc. are not consistent with the whole theme of the website.
> >   image
> > * The color of header of the **Download** page is not consistent with the whoole theme of the website.
> >   image
> > * Add dark mode feature to the website.
> 
> Sorry, but i dont know much about how to choose colors or design it. If u want u could create a new issue for it or if you have time you can provide the design and color for each elements and i will replace the color there.

ok @naveen13-06 . Leave this one. Work on the rest of the changes asked. I shall try to work on this issue later.

File: Website/pages/about.js (hunk 1 above)
Comment (SaptarshiSarkar12) [reply Website/pages/about.js:26]: ok @naveen13-06 . Leave this one. Work on the rest of the changes asked. I shall try to work on this issue later.

**Comment by SaptarshiSarkar12** [comment]
> > # Things to fix
> > 
> > * The colors of the pages like **Contribute**, etc. are not consistent with the whole theme of the website.
> >   image
> > * The color of header of the **Download** page is not consistent with the whoole theme of the website.
> >   image
> > * Add dark mode feature to the website.
> 
> Sorry, but i dont know much about how to choose colors or design it. If u want u could create a new issue for it or if you have time you can provide the design and color for each elements and i will replace the color there.

ok @naveen13-06 . I shall look into it later. I am currently looking into the Image part. Shall be reviewing it when I will be free.

File: .github/workflows/pages.yml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Simple workflow for deploying static content to GitHub Pages
 name: Deploy Drifty Website
-
+defaults:
+  run:
+    working-directory: Website
```
Comment (SaptarshiSarkar12) [inline .github/workflows/pages.yml:5]: @naveen13-06 Please remove these lines.
```suggestion
```

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@naveen13-06 Please rename *Website* folder to *New Website*. 
I shall be merging this PR and will continue with the development of the new website, myself as many things need to be added after Drifty GUI is published.

**Comment by SaptarshiSarkar12** [comment]
Hey @naveen13-06 , please move **package.json** and **package-lock.json** files inside **New Website** folder.

**Comment by naveen13-06** [comment]
> Hey @naveen13-06 , please move **package.json** and **package-lock.json** files inside **New Website** folder.

Hey @SaptarshiSarkar12  ,they are already inside New Website folder

**Comment by SaptarshiSarkar12** [comment]
> > Hey @naveen13-06 , please move **package.json** and **package-lock.json** files inside **New Website** folder.
> 
> Hey @SaptarshiSarkar12 ,they are already inside New Website folder

@naveen13-06 I think it's not. Please see below screenshot.
image

**Comment by naveen13-06** [comment]
@SaptarshiSarkar12  check now

**Comment by SaptarshiSarkar12** [comment]
@naveen13-06 Yeah, now it's okay :+1:

**Review by SaptarshiSarkar12 — APPROVED** [review]

**Comment by SaptarshiSarkar12** [comment]
@naveen13-06 Please resolve the branch conflicts else I can not merge it.

**Review by SaptarshiSarkar12 — APPROVED** [review]
@naveen13-06 **Looks good to merge 👍.** 
**Thanks for contributing 🚀 🚀.**
You may join our _Discord server_ - **https://discord.gg/DeT4jXPfkG** to get updates about the project.

**Pull request merged**